### PR TITLE
ci(ref_imgs): automatically genereate the missing reference image folders

### LINKED ISSUE #5664: When main.py is executed, the test image cannot be generated after the folder is deleted.
### Introduce the problem

When running `main.py` in tests, if the output folder is deleted, the test picture will not be generated.
 For example, delete the `tests/ref_imgs/widgets` folder, and then execute `python main.py test`. All tests pass, but the image cannot be generated. 
So can you determine whether the folder exists before generating the image, and if it does not exist, create the folder first.

### Proposal

_No response_

### PR DESCRIPTION
**Pull request by kisvegabor**
### Description of the feature or fix

fixes #5664

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.10](https://github.com/szepeviktor/astyle/releases/tag/v3.4.10) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by C47D — APPROVED** [review]

File: tests/unity/unity_support.c [hunk 1]
```diff
@@ -414,4 +418,37 @@ static void buf_to_xrgb8888(const uint8_t * buf_in, uint8_t * buf_out, lv_color_
...
+    if(pathCopy == NULL) {
+        perror("Error duplicating path");
+        exit(EXIT_FAILURE);
+    }
+
+    char * token = strtok_r(pathCopy, "/", &ptr);
+    char current_path[1024] = {'\0'}; // Adjust the size as needed
+    struct stat st;
+
+    while(token != NULL && *ptr != '\0') {
```
Comment (XuNeo) [inline tests/unity/unity_support.c]: `ptr` could become NULL at the end.

```suggestion
    while(token && ptr && *ptr != '\0') {
```

Comment (W-Mai) [reply tests/unity/unity_support.c]: I think `!= NULL` is more clear

Comment (XuNeo) [reply tests/unity/unity_support.c]: `if (pointer && pointer->elements)`
This coding style is widely used, for example in lv_indev.c. It's also common in NuttX kernel code.

For check null pointer alone, it's normally specified with != NULL explicitly. For example: `LV_ASSERT(pointer != NULL);`

Comment (kisvegabor) [reply tests/unity/unity_support.c]: Thanks, updated

**Review by XuNeo — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Pull request merged**